### PR TITLE
Update compatibility to numpy - 1.21.5

### PR DESCRIPTION
**Pull request by glitch**
Issue #670 - This updates the compatibility for numpy to a max of version 1.21.5.

`mypy` and `pytest` unit testing pass for the following numpy verisons (I tested locally as well as with the github CI runner, you can check the logs under `arkouda_tests_linux -> Build/Install Arkouda to verify which version of numpy was installed)

- 1.19.2 (https://github.com/glitch/arkouda/actions/runs/1719532839)
- 1.20.3 (https://github.com/glitch/arkouda/actions/runs/1718965418)
- 1.21.5 (https://github.com/glitch/arkouda/actions/runs/1719181717)

Note: The latest version of numpy at this time is 1.22.0 but it dropped Python 3.7.x support so I stopped with the latest 3.7.x line available in `conda` which was 1.21.5

In short the main thing which happened was deprecation warnings were added to things like `np.bool` in later versions on numpy.  The documentations said it was really just a stand in for `builtins.bool` etc.  I made updates for each of the specific items which were deprecated to use the recommended type.

Also updated was the version of `mypy` to 0.931 along with its dependency `typed-ast`.

You may need to update your local environment to the new versions of dependencies, but they should all be available with either `pip` or `conda`.

### CODE REVIEW

**Review by pierce314159 — APPROVED** [review]

**Review by hokiegeek2 — APPROVED** [review]
looks great!

File: arkouda/dtypes.py [hunk 1]
```diff
@@ -19,22 +19,23 @@
...
 
 # Union aliases used for static and runtime type checking
-bool_scalars = Union[builtins.bool,np.bool]
+bool_scalars = Union[builtins.bool, np.bool_]
 float_scalars = Union[float,np.float64]
 int_scalars = Union[int,np.int64]
-numeric_scalars = Union[float,np.float64,int,np.int64]
-numpy_scalars = Union[np.float64,np.int64,np.bool,np.uint8,np.str,np.str_]
-str_scalars = Union[str, np.str, np.str_]
+numeric_scalars = Union[float,np.float64,int,np.int64,np.uint8]
```
Comment (pierce314159) [inline arkouda/dtypes.py:33]: I just spoke with @glitch about this. This PR adds `np.uint8`  to `numeric_scalars` which seems right to me. But I wanted to double check with someone who's been around longer to make sure it's omission was an oversight and it wasn't intentionally excluded. Maybe @reuster986 or @hokiegeek2 can verify

Comment (hokiegeek2) [reply arkouda/dtypes.py:33]: @pierce314159 good question! IIRC, just an oversight

Comment (reuster986) [reply arkouda/dtypes.py:33]: yeah, I think it was just an oversight. Ideally, we would accept all numpy dtypes as scalar input (except for maybe uint64), because they're all castable to something arkouda supports, but that's a discussion for another PR :)

**Review by reuster986 — APPROVED** [review]
